### PR TITLE
Feature/200 connect GitHub account

### PR DESCRIPTION
**Pull request by SashaBondarchuk**
https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/56970649/6a3fa6f5-76da-4406-88b0-a2d8b8553fc7

https://trello.com/c/2Z6lVPp8/200-connect-github-account

### CODE REVIEW

**Review by Limbo2332 — APPROVED** [review]

**Comment by SashaBondarchuk** [comment]
[[BUG#104] [Sign Up|Sign in] Redirects the user to the landing page after registration or sing in through Google account](https://trello.com/c/xiZcWeoT/240-bug104-sign-upsign-in-redirects-the-user-to-the-landing-page-after-registration-or-sing-in-through-google-account)

File: frontend/src/app/core/services/auth.service.ts [hunk 1]
```diff
@@ -164,6 +165,35 @@ export class AuthService {
         return of(this.getUserInfo()!);
     }
 
+    linkGitHub(): Observable<firebase.auth.UserCredential | null> {
+        return this.afAuth.authState.pipe(
+            first(),
+            switchMap((user) => {
+                if (!user) {
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth.service.ts]: you can wait for first not null user in first method

File: frontend/src/app/core/services/auth.service.ts [hunk 2]
```diff
@@ -164,6 +165,35 @@ export class AuthService {
...
+                }
+
+                return user.linkWithPopup(new GithubAuthProvider()).then(
+                    (result) => result,
+                    (error) => {
+                        this.toastrNotification.showError(error.message);
+
+                        return null;
+                    },
+                );
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth.service.ts]: separate private method

File: frontend/src/app/modules/user/user-info-editor/user-info-editor.component.html [hunk 3]
```diff
@@ -57,6 +57,17 @@
                     </div>
                 </div>
             </div>
+            <p>ACCOUNTS</p>
```
Comment (tatianahutii) [inline frontend/src/app/modules/user/user-info-editor/user-info-editor.component.html:48]: could you please create a separate block (in screen we should have 2 gray blocks in black background)

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/user/user-info-editor/user-info-editor.component.html (hunk 3 above)
Comment (SashaBondarchuk) [reply frontend/src/app/modules/user/user-info-editor/user-info-editor.component.html:48]: is that okay?

image

image

**Comment by SashaBondarchuk** [comment]
[[BUG#101] [Edit profile page] The edit profile page doesn't match the design in screen resolution 360*640](https://trello.com/c/EAT9WlRR/237-bug101-edit-profile-page-the-edit-profile-page-doesnt-match-the-design-in-screen-resolution-360640)

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**